### PR TITLE
Version 6.1.0

### PR DESCRIPTION
**Pull request by JamesRobertsonGames**
multidict 6.1.0

**Destination channel:** defaults

### Links

- [PKG-5961](https://anaconda.atlassian.net/browse/PKG-5961) 
- [Upstream repository](https://github.com/aio-libs/multidict/releases/tag/v6.1.0)
- [Upstream changelog/diff](https://github.com/aio-libs/multidict/releases/tag/v6.1.0)
- Relevant dependency PRs:
  - ...

### Explanation of changes:
- Version update
- typing extensions added
- remove patch https://github.com/aio-libs/multidict/pull/864

[PKG-5961]: https://anaconda.atlassian.net/browse/PKG-5961?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by cbouss — COMMENTED** [review]
Can you build with abs.yaml to get 3.13 artifacts?
```
build_env_vars:
  ANACONDA_ROCKET_ENABLE_PY313 : yes
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,33 +1,29 @@
...
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/m/multidict/multidict-{{ version }}.tar.gz
-  sha256: 3666906492efb76453c0e7b97f2cf459b0682e7402c0489a95484965dbc1da49
-  patches:
-    # backport https://github.com/aio-libs/multidict/pull/864
-    - patches/0001-Fix-usage-python-C-API-for-version-greater-3.9-and-f.patch
```
Comment (danpetry) [inline recipe/meta.yaml:13]: patch can be deleted too

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,33 +1,29 @@
...
-  skip: true  # [py<37]
+  skip: true  # [py<38]
 
 requirements:
   build:
     - {{ compiler('c') }}
-    - patch # [unix]
-    - m2-patch # [win]
   host:
     - cython 0.29
```
Comment (danpetry) [inline recipe/meta.yaml]: I can't find any mention of cython upstream. Is it still needed?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,33 +1,29 @@
...
     - python
     - pip
     - setuptools >=40
     - wheel
+    - typing-extensions >=4.1.0 # [py<311]
```
Comment (danpetry) [inline recipe/meta.yaml]: should go in run and be removed from the test section I think

**Review by skupr-anaconda — APPROVED** [review]

**Review by cbouss — APPROVED** [review]

**Pull request merged**